pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -134,7 +143,78 @@ func NewHeadManager(logger log.Logger, dir string, metrics *Metrics, tsdbManager
 	return m
 }
 
+func (m *HeadManager) loop() {
+	defer m.wg.Done()
+
+	buildPrev := func() error {
+		if err := m.buildTSDBFromWAL(m.prev.initialized); err != nil {
+			return errors.Wrap(err, "building tsdb head")
+		}
+
+		// Now that the tsdbManager has the updated TSDBs, we can remove our references
+		m.mtx.Lock()
+		defer m.mtx.Unlock()
+		m.prevHeads = nil
+		m.prev = nil
+
+		return nil
+	}
+
+	ticker := time.NewTicker(defaultRotationCheckPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// retry tsdb build failures from previous run
+			if m.prev != nil {
+				err := buildPrev()
```

Comment:
We have metrics for rotations, but not for building TSDBs. I think we should either track both operations or combine them into the same metric.

Reply:
We still need to add metrics for when `buildTSDBFromWAL` fails.